internal/controller/processor.go

```diff
@@ -107,20 +150,22 @@ func (processor *AppConfigurationProviderProcessor) ProcessKeyValueRefresh() err
 		reconcileState.NextSentinelBasedRefreshReconcileTime = nextSentinelBasedRefreshReconcileTime
 		return nil
 	}
-
 	// Get the latest key value settings
-	updatedSettings, err := (*processor.Retriever).CreateKeyValueSettings(processor.Context, processor.ResolveSecretReference)
+	existingConfigMapSettings := &existingConfigMap.Data
+	if processor.Settings.ConfigMapSettings != nil {
+		existingConfigMapSettings = &processor.Settings.ConfigMapSettings
+	}
+	partiallyUpdatedSettings, err := (*processor.Retriever).RefreshKeyValueSettings(processor.Context, existingConfigMapSettings, processor.ResolveSecretReference)
```

Comment:
Do you mind rename `partiallyUpdatedSettings` to `keyValueRefreshedSettings`?

Reply:
Updated